symengine/ruby/ext/symengine/symengine.c

```diff
@@ -22,4 +24,11 @@ void Init_symengine() {
     rb_define_method(c_basic, "/", cbasic_div, 1);
     rb_define_method(c_basic, "**", cbasic_pow, 1);
     rb_define_method(c_basic, "-@", cbasic_neg, 0);
+    rb_define_method(c_basic, "to_s", cbasic_to_str, 0);
+    rb_define_method(c_basic, "expand", cbasic_expand, 0);
+
+    //Symbol class
+    c_symbol = rb_define_class_under(m_symengine, "Symbol", c_basic);
+    rb_define_alloc_func(c_symbol, csymbol_alloc);
```

Comment:
Okay. So, this means that each instance of `Basic` needs to have the same struct right? (because `+` is defined for `Basic` and that function expects a the sBasic)